packages/template-retail-react-app/app/components/experience/carousel/index.jsx

```diff
@@ -0,0 +1,191 @@
+ * @param {number} props.xsCarouselSlidesToDisplay - Number of children that will be rendered in view on "xs" screens.
+ * @param {number} props.smCarouselSlidesToDisplay - Number of children that will be rendered in view on "sm" screens.
+ * @param {number} props.mdCarouselSlidesToDisplay - Number of children that will be rendered in view on "md" screens.
+ * @param {Object []} props.region - The regions passed internally to this component by the `commerce-sdk-react` Page component.
+ * @returns {React.ReactElement} - Crousel component.
+ */
+const Carousel = (props = {}) => {
+    const scrollRef = useRef()
+
+    const {
+        textHeadline,
+        xsCarouselIndicators = false,
+        smCarouselIndicators = false,
+        mdCarouselIndicators = false,
+        xsCarouselControls = false,
+        smCarouselControls = false,
+        xsCarouselSlidesToDisplay = 1,
+        smCarouselSlidesToDisplay = 1,
+        mdCarouselSlidesToDisplay = 1,
+        // Internally Provided
+        regions
+    } = props
+
+    const controlDisplay = {
+        base: xsCarouselControls ? 'block' : 'none',
+        sm: xsCarouselControls ? 'block' : 'none',
+        md: smCarouselControls ? 'block' : 'none',
+        lg: 'block'
+    }
+
+    const itemWidth = {
+        base: `calc(${100 / xsCarouselSlidesToDisplay}%)`,
+        sm: `calc(${100 / xsCarouselSlidesToDisplay}%)`,
+        md: `calc(${100 / smCarouselSlidesToDisplay}%)`,
+        lg: `calc(${100 / mdCarouselSlidesToDisplay}%)`
+    }
+
+    const overflowXScroll = {
+        base: xsCarouselIndicators ? 'block' : 'none',
+        sm: xsCarouselIndicators ? 'block' : 'none',
+        md: smCarouselIndicators ? 'block' : 'none',
+        lg: mdCarouselIndicators ? 'block' : 'none'
+    }
+    const overflowXScrollValue = useBreakpointValue(overflowXScroll)
+
+    const components = regions[0]?.components || []
+    const itemCount = components.length
+
+    // Scroll the container left or right by 100%. Passing no args or `1`
+    // scrolls to the right, and passing `-1` scrolls left.
+    const scroll = useCallback((direction = 1) => {
+        scrollRef.current?.scrollBy({
+            top: 0,
+            left: (direction * window.innerWidth) / itemCount,
+            behavior: 'smooth'
+        })
+    })
+
+    // Our indicator implementation uses the scrollbar to show the context of the current
+    // item selected. Because MacOS hides scroll bars after they come to rest we need to
+    // force them to show. Please note that this feature only works on web-kit browsers,
+    // for all other brosers the scroller/indicator will be shown.
+    const css = `
+        .scroll-indicator::-webkit-scrollbar {
+            display:${overflowXScrollValue};
+            -webkit-appearance: none;
+            height: 8px;
+        }
+        .scroll-indicator::-webkit-scrollbar-thumb {
+            background-color: rgba(0, 0, 0, 0.5);
+        }
```

Comment:
As discussed IRL our design here is diverging as we have decided to base out implementation off of a previous already implemented component called the product slider. This was done to save implementation time, 1-1 parity with the UI from SFRA and PWA kit is the tradeoff. However this design, because it's based off of a pre-existing component, fits the UI better.